duckduckgo/Android
# Upgrade to Mockito 4, sort out test config problems

Task/Issue URL: https://app.asana.com/0/0/1201589003047297/f

### Description
Tidies up Mockito test dependencies, and upgrades to latest Mockito `4.2.0`

### Steps to test this PR

ℹ️  make sure you perform a gradle sync before testing in the IDE

1. Run `./gradlew checks` and verify everything passing
2. Open some test classes in IDE and check IDE not reporting any errors. e.g., `WebViewLongPressHandlerTest`, which shouldn't show an error on `openMocks` usage.
